R/aorsf_data.R

```diff
@@ -0,0 +1,225 @@
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "trees",
+    original = "n_tree",
+    func = list(pkg = "dials", fun = "trees"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "min_n",
+    original = "leaf_min_obs",
+    func = list(pkg = "dials", fun = "min_n"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "rand_forest",
+    eng = "aorsf",
+    parsnip = "mtry",
+    original = "mtry",
+    func = list(pkg = "dials", fun = "mtry"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+  parsnip::set_fit(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    value = list(
+      interface = "formula",
+      protect = c("formula", "data", "weights"),
+      func = c(pkg = "aorsf", fun = "orsf"),
+      defaults =
+        list(
+          n_thread = 1,
+          verbose_progress = FALSE
+        )
+    )
+  )
+
+  parsnip::set_encoding(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "regression",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "class",
+    value = list(
+      pre = NULL,
+      # makes prob preds consistent with class ones
+      post = function(results, object){
+
+        missings <- apply(results, 1, function(x) any(is.na(x)))
+
+        if(!any(missings))
+          return(colnames(results)[apply(results, 1, which.max)])
+
+        obs <- which(!missings)
+
+        out <- rep(NA_character_, nrow(results))
+        out[obs] <- colnames(results)[apply(results[obs, ], 1, which.max)]
+        out
+
+      },
+      func = c(fun = "predict"),
+      args =
+        list(
+          object = quote(object$fit),
+          new_data = quote(new_data),
+          pred_type = "prob",
+          verbose_progress = FALSE,
+          na_action = 'pass'
+        )
+    )
+  )
+
+  parsnip::set_pred(
+    model = "rand_forest",
+    eng = "aorsf",
+    mode = "classification",
+    type = "prob",
+    value = list(
+      pre = NULL,
+      post = function(x, object) {
+        as_tibble(x)
+      },
```

Comment:
```suggestion
      post = NULL,
```

Correct me if I'm wrong, but I _believe_ this should give the same output. `ensure_parsnip_format()` should take care of the tibble conversion, and not doing so twice will avoid performance overhead from duplicate `as_tibble()` calls.

Reply:
Good catch! I'll pull this down and see. I'm confident you're right.

Reply:
I encountered an error without the post application of `as_tibble()`. Not sure exactly what went wrong:
![image](https://github.com/tidymodels/bonsai/assets/11821781/9585fb4d-fe16-4b66-9660-e834ae741ae4)

Reply:
Heard. `as_tibble()` works, then!